library/Icinga/Crypt/AesCrypt.php

```diff
@@ -217,4 +232,35 @@ public function encryptToBase64($data)
     {
         return base64_encode($this->encrypt($data));
     }
+
+    private function getDecryptCBC($data)
+    {
+        $c = base64_decode($data);
+        $ivlen = openssl_cipher_iv_length($this->method);
+        $iv = substr($c, 0, $ivlen);
+        $hmac = substr($c, $ivlen, $sha2len=32);
+        $data = substr($c, $ivlen + $sha2len);
+
+        $decrypt = openssl_decrypt($data, $this->method, $this->getKey(), 0, $this->getIV());
+        $calcmac = hash_hmac('sha256', $data, $this->getKey(), $as_binary=true);
+
+        if ((is_bool($decrypt) && $decrypt === false) || ! hash_equals($hmac, $calcmac)) {
+            throw new RuntimeException('Decryption failed');
+        }
+
+        return $decrypt;
+    }
+
+    private function getEncryptCBC($data)
+    {
+        $encrypt = openssl_encrypt($data, $this->method, $this->getkey(), 0, $this->getIV());
+
+        if (is_bool($encrypt) && $encrypt === false) {
+            throw new RuntimeException('Encryption failed');
+        }
+
+        $hmac = hash_hmac('sha256', $encrypt, $this->getkey(), $as_binary=true);
+
+        return base64_encode($this->getIV() . $hmac . $encrypt);
+    }
```

Comment:
* Please don't define variables in calls (e.g. `$sha2len` and `$as_binary`, the latter even doesn't use camel case notation)
* You can drop the `is_bool` check when using the strict comparison operator for the variable in question. Strict comparison also checks the type.
* Why is there `$iv` and `$this->getIV()`? Are both the same? Can't `$data` have only the `$hmac` if the IV is already known otherwise?
* Please remove the `get` prefix from both methods, they're no getters